pkg/console/operator/sync_v400.go

```diff
@@ -392,6 +400,74 @@ func (co *consoleOperator) SyncRoute(operatorConfig *operatorv1.Console) (*route
 	return rt, rtIsNew, rtErr
 }
 
+func (c *consoleOperator) SyncCustomLogoConfigMap(operatorConfig *operatorsv1.Console) (okToMount bool, err error) {
+	// validate first, to avoid a broken volume mount & a crashlooping console
+	okToMount, err = c.ValidateCustomLogo(operatorConfig)
+
+	if okToMount {
+		if err := c.UpdateCustomLogoSyncSource(operatorConfig); err != nil {
+			klog.V(4).Infoln("custom logo sync source update error")
+			return false, customerrors.NewCustomLogoError("custom logo sync source update error")
+		}
+	}
+	return okToMount, err
+}
+
+// on each pass of the operator sync loop, we need to check the
+// operator config for a custom logo.  If this has been set, then
+// we notify the resourceSyncer that it needs to start watching this
+// configmap in its own sync loop.  Note that the resourceSyncer's actual
+// sync loop will run later.  Our operator is waiting to receive
+// the copied configmap into the console namespace for a future
+// sync loop to mount into the console deployment.
+func (c *consoleOperator) UpdateCustomLogoSyncSource(operatorConfig *operatorsv1.Console) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+
+	if logoConfigMapName != "" {
+		source.Name = logoConfigMapName
+		source.Namespace = api.OpenShiftConfigNamespace
+	}
+	// if no custom logo provided, sync an empty source to delete
+	return c.resourceSyncer.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Namespace: api.OpenShiftConsoleNamespace, Name: api.OpenShiftCustomLogoConfigMapName},
+		source,
+	)
+}
+
+func (c *consoleOperator) ValidateCustomLogo(operatorConfig *operatorsv1.Console) (okToMount bool, err error) {
```

Comment:
This is fairly thorough, but one thing to note.  The operator will loop on an error until it is corrected as we are deliberately avoiding a crashloop on the console pods.  We aren't bringing the operand to the desired state if the user does the config incorrectly, however, we are reporting on the clusteroperator with specific errors.  This should be easy for a user to correct if they get in a bad state.

Reply:
make sense